common/utils/src/main/java/org/apache/spark/internal/Logger.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.apache.logging.log4j.CloseableThreadContext;
+import org.apache.logging.log4j.message.MessageFactory;
+import org.apache.logging.log4j.message.ParameterizedMessageFactory;
+
+public class Logger {
+
+  private static final MessageFactory MESSAGE_FACTORY = ParameterizedMessageFactory.INSTANCE;
+  private final org.slf4j.Logger slf4jLogger;
+
+  Logger(org.slf4j.Logger slf4jLogger) {
+    this.slf4jLogger = slf4jLogger;
+  }
+
+  public void error(String msg) {
+    slf4jLogger.error(msg);
+  }
+
+  public void error(String msg, Throwable throwable) {
+    slf4jLogger.error(msg, throwable);
+  }
+
+  public void error(String msg, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.error(msg);
+    } else {
+      withLogContext(msg, mdcs, null, mt -> slf4jLogger.error(mt.message));
+    }
+  }
+
+  public void error(String msg, Throwable throwable, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.error(msg, throwable);
+    } else {
+      withLogContext(msg, mdcs, throwable, mt -> slf4jLogger.error(mt.message, mt.throwable));
+    }
+  }
+
+  public void warn(String msg) {
+    slf4jLogger.warn(msg);
+  }
+
+  public void warn(String msg, Throwable throwable) {
+    slf4jLogger.warn(msg, throwable);
+  }
+
+  public void warn(String msg, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.warn(msg);
+    } else {
+      withLogContext(msg, mdcs, null, mt -> slf4jLogger.warn(mt.message));
```

Comment:
Let's check the log level before calling `withLogContext`. For example, if the log level is ERROR, we don't need to enter the method `withLogContext`

Reply:
Done